rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/topic_cache.hpp

```diff
@@ -89,9 +108,16 @@ class TopicCache
   /**
    * @return a map of topic name to the vector of topic types used.
    */
-  const TopicToTypes & getTopicToTypes() const
+  const TopicToTypes getTopicToTypes() const
```

Comment:
`TopicToTypes` is now inferred from `TopicNameToTopicData` and is not stored as is in a variable.

Reply:
That's fine. Adding overhead is always unfortunate, but I don't have evidence to suggest we should duplicate data to speed this up.